### PR TITLE
VFK: Support for UTF-8

### PR DESCRIPTION
**Pull request by isax-sro**
## What does this PR do?

Support to UTF-8 data

## Environment

Provide environment details, if relevant:

* OS: Ubuntu 20.04
* Compiler: gcc

### CODE REVIEW

**Comment by rouault** [comment]
Does this need to be backported to 3.5 branch ?

File: ogr/ogrsf_frmts/vfk/vfkreader.cpp [hunk 1]
```diff
@@ -154,6 +154,65 @@ char *VFKReader::ReadLine()
...
+            break;
+        }
+        if (pszLine[1] != 'H') {
+            /* (not) 'H' header */
+            CPLFree(pszLine);
+            continue;
+        }
+
+        char *poKey = pszLine + 2; /* &H */
+        char *poValue = poKey;
```
Comment (rouault) [inline ogr/ogrsf_frmts/vfk/vfkreader.cpp:185]: Naming convention: this should pszKey and pszValue

**Comment by isax-sro** [comment]
Hello,
I think it would be very useful as from the 14th April 2022 data from 
state institution are exported only in UTF-8. But Martin Landa should 
confirm this.

Michal

**Comment by nyalldawson** [comment]
The layer should also return true for the OLCStringsAsUTF8 capability now, shouldn't it?

**Comment by rouault** [comment]
note that if this needs to go in 3.5.1 (@landam  is that needed ? or is your previous commit in the driver sufficient?), this PR should be finalized to take into account above comments before June 27th

**Review by landam — APPROVED** [review]

**Comment by landam** [comment]
> note that if this needs to go in 3.5.1 (@landam is that needed ? or is your previous commit in the driver sufficient?), this PR should be finalized to take into account above comments before June 27th

This PR also contains important fix for octal representation of the currenncy sign. @rouault Therefore I think it should be backported to 3.5 branch.

**Comment by rouault** [comment]
My and Nyall's comment need to be taken into account before being able to merge this

**Comment by landam** [comment]
> My and Nyall's comment need to be taken into account before being able to merge this

@isax-sro Please change your PR accordingly.

**Comment by isax-sro** [comment]
Hello, what do you mean "change your PR accordingly"? Any hint? I am not 
familiar with github/GDAL workflow.

**Comment by landam** [comment]
> nevermind: as time is ticking and I want to issue a release candidate today, I've just pushed myself the fixes

Thanks @rouault !

**Pull request merged**